src/backend/distributed/commands/drop_distributed_table.c

```diff
@@ -20,21 +20,45 @@
 #include "utils/lsyscache.h"
 
 
+/* local function forward declarations */
+static void MasterDropDistributedTableMetadataOnWorkers(Oid relationId, char *schemaName,
+														char *tableName);
+
+
 /* exports for SQL callable functions */
 PG_FUNCTION_INFO_V1(master_drop_distributed_table_metadata);
+PG_FUNCTION_INFO_V1(master_remove_partition_metadata);
+PG_FUNCTION_INFO_V1(master_drop_distributed_table_metadata_on_workers);
 
 
 /*
- * master_drop_distributed_table_metadata removes the entry of the specified distributed
- * table from pg_dist_partition and drops the table from the workers if needed.
+ * master_drop_distributed_table_metadata UDF is a stub UDF to install Citus flawlessly.
+ * Otherwise we need to delete them from our sql files, which is confusing and not a
+ * common operation in the code-base.
+ *
+ * This function is basically replaced with
+ * master_drop_distributed_table_metadata_on_workers() followed by
+ * master_remove_partition_metadata().
  */
 Datum
 master_drop_distributed_table_metadata(PG_FUNCTION_ARGS)
+{
+	ereport(DEBUG2, (errmsg("this function is deprecated and no longer is used")));
+
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * master_remove_partition_metadata removes the entry of the specified distributed
+ * table from pg_dist_partition.
+ */
+Datum
+master_remove_partition_metadata(PG_FUNCTION_ARGS)
```

Comment:
I did not like `partition` here. It could be confused with logical partitioning.

Reply:
I wonder if we can merge this functionality with `master_drop_all_shards`. Since these `master_drop_distributed_table_metadata_on_workers` and `master_remove_partition_metadata` are used with that function

Reply:
> I did not like partition here. It could be confused with logical partitioning.

Well, the metadata table is `pg_dist_partition`. I think that's the main issue we should consider fixing in the future 😄  If you have strong opinions on this, feel free to propose a name for the function.

Also, many people refer to partitioning as [`declarative partitioning`](https://www.postgresql.org/docs/10/static/ddl-partitioning.html) in the Postgres community. So, should be OK?

> I wonder if we can merge this functionality with master_drop_all_shards. Since these master_drop_distributed_table_metadata_on_workers and master_remove_partition_metadata are used with that function

I prefer to keep the functions separately for two reasons: 
- This seems easier/clearer to read since otherwise, the reader might have harder time to understand where the `pg_dist_partition` entry is removed
- The flow of the code makes sense: (i) `DROP table` drops the table on the coordinator and that's why we're on the trigger here (ii) Drop the table and the corresponding metadata on the MX workers -- if the cluster is MX (iii) drop the shards (iv) remove `pg_dist_partition` entry on the coordinator